src/main/assisted-installer-controller-ocp/main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/openshift/assisted-installer/src/k8s_client"
+	"github.com/openshift/assisted-installer/src/utils"
+
+	"github.com/kelseyhightower/envconfig"
+	assistedinstallercontroller "github.com/openshift/assisted-installer/src/assisted_installer_controller"
+	"github.com/openshift/assisted-installer/src/inventory_client"
+	"github.com/openshift/assisted-installer/src/ops"
+	"github.com/sirupsen/logrus"
+)
+
+var Options struct {
+	ControllerConfig assistedinstallercontroller.ControllerConfig
+}
+
+func main() {
+	logger := logrus.New()
+
+	err := envconfig.Process("myapp", &Options)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	kc, err := k8s_client.NewK8SClient("", logger)
+	if err != nil {
+		log.Fatalf("Failed to create k8 client %v", err)
+	}
+
+	logger.Infof("Start running assisted-controller with cluster-id %s, url %s",
+		Options.ControllerConfig.ClusterID, Options.ControllerConfig.URL)
+
+	err = kc.SetProxyEnvVars()
+	if err != nil {
+		log.Fatalf("Failed to set env vars for installer-controller pod %v", err)
+	}
+
+	client, err := inventory_client.CreateInventoryClient(Options.ControllerConfig.ClusterID,
+		Options.ControllerConfig.URL, Options.ControllerConfig.PullSecretToken, Options.ControllerConfig.SkipCertVerification,
+		Options.ControllerConfig.CACertPath, logger, utils.ProxyFromEnvVars)
+	if err != nil {
+		log.Fatalf("Failed to create inventory client %v", err)
+	}
+
+	assistedController := assistedinstallercontroller.NewController(logger,
+		Options.ControllerConfig,
+		ops.NewOps(logger, false),
+		client,
+		kc,
+	)
+
+	logger.Infof("Controller deployed on OCP cluster")
+
+	for {
+		var wg sync.WaitGroup
+		ctxApprove, cancelApprove := context.WithCancel(context.Background())
+		go assistedController.ApproveCsrs(ctxApprove, &wg)
+		wg.Add(1)
```

Comment:
this will surely work, but i would suggest that you create a specific goroutine function per ApproveCsr and nodeState, and the loop will be inside each of those functions. This way approveCSR won't have to wait until waitAndUpdate finished in order to be run again

Reply:
Moved to two separate goroutines

Reply:
Why is this required (approving CSR), why not let cluster-machine-approver handle it?

Reply:
because currently we don't provide BMHs in day2. Once we wil start using BMHs (probably 4.7),we can skip approve CSR